src/main/java/com/ericsson/ei/config/ConfigurationLogger.java

```diff
@@ -1,78 +1,71 @@
-package com.ericsson.ei.config;
+/*
+   Copyright 2019 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
 
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.ei.config;
 
-import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.core.env.Environment;
-import org.springframework.stereotype.Component;
+import org.springframework.boot.context.event.ApplicationPreparedEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.MapPropertySource;
+import org.springframework.core.env.PropertySource;
 
-@Component
-public class ConfigurationLogger {
+public class ConfigurationLogger implements ApplicationListener<ApplicationPreparedEvent> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ConfigurationLogger.class);
 
-    @Autowired
-    private Environment environment;
+    private ConfigurableEnvironment environment;
 
-    @PostConstruct
-    public void print() {
+    @Override
+    public void onApplicationEvent(ApplicationPreparedEvent event) {
```

Comment:
When is this run? Will it be on startup as we have today with PostConstruct?

Reply:
This runs and prints almost immediately after application starts. It is in fact much faster than before when it was a component and could take around 3-5s before you got the printout.

Reply:
Oh that sounds promising!

Reply:
And ofc we don't need to manually list every property. =)